src/test/java/org/broadinstitute/consent/http/service/DataAccessRequestServiceTest.java

```diff
@@ -782,11 +750,199 @@ void testValidateNoKeyPersonnelDuplicatesSO() {
+    when(institutionService.findInstitutionForEmail(badEmailAddress)).thenReturn(null);
+    initService();
+    assertThrows(IllegalArgumentException.class, () -> service.validatePersonnelInSameInstitution(user, data));
+  }
+
+  @Test
+  void testValidatePersonnelInSameInstitutionThrowsForBadCollaborator() {
+    String goodEmailAddress = "j@example.com";
+    String badEmailAddress = "bad@evil.com";
+    User user = new User(1, "j@example.com", "Display Name", new Date());
+    Institution goodInstitution = new Institution();
+    goodInstitution.setId(1);
+    user.setInstitution(goodInstitution);
+    DataAccessRequestData data = new DataAccessRequestData();
+    data.setPiEmail(goodEmailAddress);
+    data.setSigningOfficialEmail(goodEmailAddress);
+    data.setItDirectorEmail(goodEmailAddress);
+    Collaborator collaborator1 = new Collaborator();
+    collaborator1.setEmail(goodEmailAddress);
+    Collaborator collaborator2 = new Collaborator();
+    collaborator2.setEmail(badEmailAddress);
+    data.setLabCollaborators(List.of(collaborator1, collaborator2));
+    when(institutionService.findInstitutionForEmail(goodEmailAddress)).thenReturn(goodInstitution);
+    when(institutionService.findInstitutionForEmail(badEmailAddress)).thenReturn(null);
+    initService();
+    assertThrows(IllegalArgumentException.class, () -> service.validatePersonnelInSameInstitution(user, data));
+  }
+
+  @Test
+  void testValidatePersonnelInSameInstitutionThrowsForBadLabStaffMember() {
+    String goodEmailAddress = "j@example.com";
+    String badEmailAddress = "bad@evil.com";
+    User user = new User(1, "j@example.com", "Display Name", new Date());
+    Institution goodInstitution = new Institution();
+    goodInstitution.setId(1);
+    user.setInstitution(goodInstitution);
+    DataAccessRequestData data = new DataAccessRequestData();
+    data.setPiEmail(goodEmailAddress);
+    data.setSigningOfficialEmail(goodEmailAddress);
+    data.setItDirectorEmail(goodEmailAddress);
+    Collaborator collaborator1 = new Collaborator();
+    collaborator1.setEmail(goodEmailAddress);
+    Collaborator collaborator2 = new Collaborator();
+    collaborator2.setEmail(goodEmailAddress);
+    data.setLabCollaborators(List.of(collaborator1, collaborator2));
+    Collaborator labStaffMember = new Collaborator();
+    labStaffMember.setEmail(goodEmailAddress);
+    Collaborator labStaffMember2 = new Collaborator();
+    labStaffMember2.setEmail(badEmailAddress);
+    data.setLabCollaborators(List.of(labStaffMember, labStaffMember2));
+    when(institutionService.findInstitutionForEmail(goodEmailAddress)).thenReturn(goodInstitution);
+    when(institutionService.findInstitutionForEmail(badEmailAddress)).thenReturn(null);
+    initService();
+    assertThrows(IllegalArgumentException.class, () -> service.validatePersonnelInSameInstitution(user, data));
+  }
+
+  @Test
+  void testValidatePersonnelInSameInstitution_NoCollaborators() {
```

Comment:
This code is testing each case separately but since the.code tests all cases at the same time, you could combine and write a test that checks for all invalid cases at once. Or, have one setup that creates the data will all the different cases, and have separate tests that assert each case in the error message.